src/ros_video_components/include/ros_video_components/ros_battery_indicator.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef ROS_BATTERY_INDICATOR_HPP
+#define ROS_BATTERY_INDICATOR_HPP
+
+//QT
+#include <QQuickPaintedItem>
+#include <QPainter>
+
+//ROS
+#include <ros/ros.h>
+#include <std_msgs/Float32.h>
+
+
+class  ROS_Battery_Indicator: public QQuickPaintedItem {
+    //make this a Qt Widget
+    Q_OBJECT
+    // defines a qml value for the topic
+    Q_PROPERTY(QString topic READ get_topic WRITE set_topic NOTIFY topic_changed)
+
+    public:
+        // Constructor, takes parent widget, which defaults to null
+        ROS_Battery_Indicator(QQuickItem * parent = 0);
+
+        void paint(QPainter *painter);
+        void setup(ros::NodeHandle * nh);
+
+        //getters and setters
+        void set_topic(const QString &new_value);
+        QString get_topic() const;
+
+    signals:
+        void topic_changed();
+
+    private:
+        void receive_signal(const std_msgs::Float32::ConstPtr & msg);
```

Comment:
`receive_signal` is not a proper variable name here. I named it `receive_signal` because it was literally receiving the signal strength of the antenna. For you, it would be something like `receive_battery_voltage`.